sklearn/covariance/_empirical_covariance.py

```diff
@@ -84,7 +84,7 @@ def empirical_covariance(X, *, assume_centered=False):
            [0.25, 0.25, 0.25],
            [0.25, 0.25, 0.25]])
     """
-    X = np.asarray(X)
+    X = check_array(X, ensure_2d=False)
```

Comment:
The default `force_all_finite=True` will introduce overhead in a few places. For example:

https://github.com/scikit-learn/scikit-learn/blob/00032b09c7feea08edd4486c522c2d962f9d52ec/sklearn/covariance/_empirical_covariance.py#L239-L244

In the above snippet, `X` is already validated in `self._validate_data`, and calling `empirical_covariance` will check for finite-ness again.

A work around is to introduce `with config_context(assume_finite=True)` in places where we can assume the input is finite. In the above example:

```python
with config_context(assume_finite=True):
    covariance = empirical_covariance(X, assume_centered=self.assume_centered)
```